Adding devcontainer.json and Dockerfile to ease development in VSCode

GSNCodes opened the pull request:
Have added support to use devcontainers in VSCode to test out the ROS MCP Server with ease.
Relevant details can be found here: #93 

Please comment on your thoughts regarding the design and especially the edits to the README file.

Cheers!


#### Tested with TurtleSim
<img width="2485" height="1253" alt="testOutput" src="https://github.com/user-attachments/assets/f6dff423-16ab-4224-a1f0-ffd95152a00c" />

rjohn-v commented on README.md:
```diff
@@ -86,6 +87,27 @@ Follow the [installation guide](docs/installation.md) for step-by-step instructi
 4. Configure your client to connect to the ROS MCP Server  
 5. Start `rosbridge` on the target robot  
 
+**Optional: Using Devcontainer**  
```
These should definitely be in installation.md, to keep the main readme clean.

(Will test the devcontainer and give a more in depth review, but getting this comment out to you fist. )

GSNCodes replied:
Sounds good. I've moved the steps to the installation.md as suggested :smile:

GSNCodes commented:
Hey @stex2005 
Thank you for the valuable comments. I've updated the files to add support for `WSL`. Could you help me test this as I do not have access to a windows machine at the moment?

I've also updated the doc to indicate that this setup currently works only on Ubuntu, once we verify with your testing we can update that to include Windows as well.

GSNCodes commented on .devcontainer/devcontainer.json:
```diff
@@ -0,0 +1,37 @@
+      ]
+    }
+  },
+  "postStartCommand": "chmod 700 /root/.ssh && chmod 600 /root/.ssh/config /root/.ssh/id_* && chown -R root:root /root/.ssh"
```
Have removed them and made the config cross-platform compatible :+1:

GSNCodes commented on .devcontainer/devcontainer.json:
```diff
@@ -0,0 +1,37 @@
+    "--env=DISPLAY",
+    "--env=QT_X11_NO_MITSHM=1",
+    "--volume=/tmp/.X11-unix:/tmp/.X11-unix:rw",
+    "--device=/dev/dri"
```
Have removed this as well :+1:

rjohn-v commented on docs/installation.md:
```diff
@@ -317,3 +317,66 @@ What topics and services do you see on the robot?
+     </details>
+     
+
+4. You can now control the Turtlesim robot following **Step-2**, **Step-3.2** and **Step-4** from the above given manual installation.
```
Tested with VS Code and WSL. The devcontainer worked well and I was able to launch the server using uv. (So this would work well if we configure it with the http transport layer. )

However, if we control the turtlesim using the steps described here, then my understanding is that Claude Desktop is bypassing the devcontainer to test and run the server. So the only things running inside the devcontainer is the ROS instance and ROS bridge? 

This is an excellent feature to give a test ROS environment, though I think we should be clear about what we are providing to the user here.

rjohn-v replied:
My understanding is that (And correct me if I'm wrong here)

For stdio transport:
- The devcontainer gives us a containerized ROS instance with rosbridge installed, bypassing step 3.1
- Hopefully we would be able to launch rosbridge as well on startup of the container, thus bypassing step 3 entirely.

For http transport we have a lot more:
- Everything that we had for stdio
- Launching the server within the container, so skipping the need for uv as well. 

We've not yet updated the installation docs to show http transport, so (if my understanding is right), lets make this installation doc a bit more clear about what we expect the user to do?

stex2005 replied:
Thanks for clarifying the current setup. My main concern is that a devcontainer which only spins up a ROS/rosbridge instance feels a bit out of scope for this repository. From a user’s perspective, it could be confusing if the container is positioned as a full test environment but in reality is only providing part of the stack.

To make the devcontainer more aligned with this project’s purpose, I would suggest that we at least include the HTTP transport variant of the MCP server within the container, so that users can launch both the ROS environment and the MCP transport layer directly inside the devcontainer. That way, the container provides a self-contained, representative environment for the repo rather than just being a generic ROS sandbox.

GSNCodes replied:
Hey @stex2005 
Yup, already on board with that. All I meant was that currently, I have not tested the changes with the HTTP layer. Support for that is already in the works and my next commit would include it. Once that's done, I'll update the installation doc to be as specific as possible.

GSNCodes replied:
@stex2005 Were you able to test the devcontainer changes I pushed yesterday?

rjohn-v reviewed (changes requested):
I was able to run the devcontainer from WSL yesterday (before the changes requested by @stex2005) It did take more than 10 minutes to startup on my travel machine without a gpu :) . 

More importantly, the instructions of having the Devcontainer as 'Option 2' but then having it point back to everything except step 3.1 of 'option 1' was a bit confusing. (I've left some more comments on the line itself)

The instruction page is an important waypoint (and turn-off point) for potential users of the repo. We need to make sure that it is understandable at all times!
